platform/ui/src/components/studyList/TableSearchFilter.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { isInclusivelyBeforeDay } from 'react-dates';
+import CustomDateRangePicker from './CustomDateRangePicker.js';
+import { Icon } from './../../elements/Icon';
+
+function TableSearchFilter(props) {
+  const {
+    meta,
+    values,
+    onSort,
+    onValueChange,
+    sortFieldName,
+    sortDirection,
+  } = props;
+  const [focusedInput, setFocusedInput] = useState(null);
+
+  const sortIcons = ['sort', 'sort-up', 'sort-down'];
+  const sortIconForSortField =
+    sortDirection === 'asc' ? sortIcons[1] : sortIcons[2];
+  //
+  const today = moment();
+  const lastWeek = moment().subtract(7, 'day');
+  const lastMonth = moment().subtract(1, 'month');
+  // TODO: Does this need to be preserved? Default sort could be nice for large data sources
+  // this.props.studyListDateFilterNumDays,
+  const defaultStartDate = moment().subtract(365 * 10, 'days');
+  const defaultEndDate = today;
+  const studyDatePresets = [
+    {
+      text: 'Today',
```

Comment:
This label can be translated?

Reply:
I've updated to use translation; although, I think this is the first functional component that does so. It's also my first time messing with them outside of some light research. Can you double check my implementation?